lib/eco/debug.py

```diff
@@ -13,22 +13,23 @@ def __init__(self, window):
         # incoming signals
         self.signal_command = SIGNAL("command")
         self.signal_start = SIGNAL("start")
-        # outcoming signals
+        # outgoing signals
         self.signal_done = SIGNAL("finished")
         self.signal_output = SIGNAL("output")
         self.signal_toggle_buttons = SIGNAL("disablebuttons")
+        self.signal_execute_fail = SIGNAL("executefail")
         self.telnet = None
 
     def start_pdb(self):   
         self.breakpoints = {'keep': [], 'del': []}      
         self.line_read_until = '(Pdb)'   
         self.process = self.window.getEditor().tm.export(debug=True)
-        flags = fcntl(self.process.stdout, F_GETFL)
-        fcntl(self.process.stdout, F_SETFL, flags | O_NONBLOCK)       
         if not self.process:
-            self.emit(self.signal_output, "Cannot run debugger process, cannot debug.")
+            self.emit(self.signal_execute_fail)
             self.exit()
             return
+        flags = fcntl(self.process.stdout, F_GETFL)
```

Comment:
I've never spent much time on eco, so here's some dumb questions:

Why does popping up an error require changes to file descriptor flags?